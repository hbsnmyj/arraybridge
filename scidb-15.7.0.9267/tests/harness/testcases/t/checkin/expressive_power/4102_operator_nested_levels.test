# ticket:4102, allow 95 nested levels of AFL operators, but not 96.
--setup
--test

# 95 operators (94 project + 1 build) is fine.
project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(build(<a:double>[i=0:7,1,0],0),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a)

# 96 operators is not fine.
--error --code=scidb::SCIDB_SE_PARSER::SCIDB_LE_QUERY_HAS_TOO_DEEP_NESTING_LEVELS "project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(build(<a:double>[i=0:7,1,0],0),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a)"

--cleanup
