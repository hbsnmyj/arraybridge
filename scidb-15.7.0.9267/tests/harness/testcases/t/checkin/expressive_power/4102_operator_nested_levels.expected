{i} a
{0} 0
{1} 0
{2} 0
{3} 0
{4} 0
{5} 0
{6} 0
{7} 0

[An error expected at this place for the query "project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(project(build(<a:double>[i=0:7,1,0],0),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a),a)". And it failed with error code = scidb::SCIDB_SE_PARSER::SCIDB_LE_QUERY_HAS_TOO_DEEP_NESTING_LEVELS. Expected error code = scidb::SCIDB_SE_PARSER::SCIDB_LE_QUERY_HAS_TOO_DEEP_NESTING_LEVELS.]

